server/controllers/items/recent_public.coffee

```diff
@@ -0,0 +1,40 @@
+__ = require('config').universalPath
+_ = __.require 'builders', 'utils'
+error_ = __.require 'lib', 'error/error'
+items_ = __.require 'controllers', 'items/lib/items'
+user_ = __.require 'controllers', 'user/lib/user'
+
+module.exports = (req, res)->
+  { query } = req
+  assertImage = query['assert-image'] is 'true'
+  reqUserId = req.user?._id
+
+  items_.publicByDate 100, 0, assertImage, reqUserId
+  .then selectRecentItems
+  .then bundleOwnersData.bind(null, res, reqUserId)
+  .catch error_.Handler(req, res)
+
+selectRecentItems = (items)->
+  itemsLimit = 15
+  maxItemsPerOwner = 3
+
+  Promise.resolve _.groupBy items, (item)-> item.owner
+  .then (itemsByOwner)->
+    _.map itemsByOwner, (items)->
+      items.slice 0, maxItemsPerOwner
+  .then (limitedItemsByOwner)->
+    _.flatten _.values limitedItemsByOwner
+  .then (allRecentItems)->
+    allRecentItems.slice 0, itemsLimit
+
+bundleOwnersData = (res, reqUserId, items)->
```

Comment:
for functions that are duplicated with [./last_public](https://github.com/inventaire/inventaire/blob/0cf55dc/server/controllers/items/last_public.coffee), you can either extract the concerned functions to a third module both controllers would require, or we could consider that `recent-public` makes `last-public` obsolete and `git rm` it

Reply:
as `last_public` will not be used, dead code can be removed
git archives it anyway